regex/hours.regex

```diff
@@ -1 +1 @@
-([0-9]{4}[AP]M)-([0-9]{4}[AP]M)
+([0-9]{4}[AP]M)?-([0-9]{4}[AP]M)?
```

Comment:
@ammongit 
Added the case for a partially missing time like `-1200PM` and `0100PM-` in commit:

98791da7a4fe82e9cd1f734fcfa08445f87db737